### DISCUSSION TITLE
SMTP not sending emails

### DISCUSSION THREAD

**Discussion by Vincent-BePressure** [Leantime/leantime#629] [Q&A] [answered]
I am using an SMTP server with no auth, no TLS, port 25.  Leantime isn't sending emails. 

I edited line 80 in class.mailer.php to false, but that didn't seem to help. 

There is a lack of 'test email' functionality, how do we debug?  I am trying the 'forgot password' button, but can't find any logs and receive no errors.

**Comment by aboyington** [comment]
Has anyone been able to get SMTP working or able to send emails? if so how?

**Comment by marcelfolaron** [comment, marked as answer]
Hi,

Can you set debug to true in core/class.mailer.php line 76?

This should give you some additional smtp output

**Reply by aboyington** [reply]
Thank you, it looks like I was missing the from field in the email settings and I edited line 80 in class.mailer.php so emails are being sent using phpmailer instead of SMTP, which is good enough for now.

**Comment by leogaggl** [comment]
I do agree with OP @Vincent-BePressure that some kind of "Test Email" function would be great to assist in debugging.

It would be a common issue with all the different settings that various SMTP providers need.

Great feature request!

**Comment by d3corator** [comment]
+1

**Comment by martinkoutecky** [comment]
Hi - I was still not able to get email working, and now I'm confused by the configuration. I'm using docker-compose. I'm setting the following variables (relevant to email):

```
      LEAN_EMAIL_RETURN: 'leantime@domain.com'
      LEAN_EMAIL_USE_SMTP: true
      LEAN_EMAIL_SMTP_HOSTS: 'domain.com'
      LEAN_EMAIL_SMTP_USERNAME: 'leantime@domain.com'
      LEAN_EMAIL_SMTP_PASSWORD: 'somepassword'
      LEAN_EMAIL_SMTP_SECURE: 'SSL'
      LEAN_EMAIL_SMTP_PORT: 465
```

However I'm confused whether those are the right variables names given that [this](https://github.com/tiredofit/docker-leantime/tree/main#environment-variables) mentions variables like `SMTP_ENABLE`, `SMTP_HOST`, etc. Am I even setting the right variables? How could I enable some debug to get more information in the logs? Etc.

(The way I'm testing now is the "forgot password" link, and after filling in the email, the website times out with 504 bad gateway error. Nothing relevant appears in the logs.)

**Reply by martinkoutecky** [reply]
Hm, seems like I fixed it by changing `LEAN_EMAIL_SMTP_PORT: 465` to `587`. This is counterintuitive because I'm using [mailcow](https://mailcow.email/) and the [manual configuration](https://docs.mailcow.email/client/client-manual/) page states that for `SSL` one should use port `465`. But, it works, after all.

**Reply by gamedevkirk** [reply]
This worked for me as well but it feels super hacky.

**Comment by gmignotte** [comment]
Hi, I don't receive any email and I don't know how I can debug this ?

My configuration : 
```
    /* Email */
    public $email = 'mailer@domain.com';		//Return email address
    public $useSMTP = true;					//Use SMTP? If set to false, the default php mail() function will be used
    public $smtpHosts = 'mail.infomaniak.com';		//SMTP host
    public $smtpAuth = true;					//SMTP use user/password authentication
    public $smtpUsername = 'mailer@domain.com';	//SMTP username
    public $smtpPassword = 'xxxxxxxxxxxx';	                //SMTP password
    public $smtpAutoTLS = true;				//SMTP Enable TLS encryption automatically if a server supports it
    public $smtpSecure = 'SSL';					//SMTP Security protocol (usually one of: TLS, SSL, STARTTLS)
    public $smtpSSLNoverify = true;				//SMTP Allow insecure SSL: Don't verify certificate, accept self-signed, etc.
    public $smtpPort = '465';					//Port (usually one of 25, 465, 587, 2526)
```
I've tried many configuration without success and I haven't anything in the log file.
In my DNS I have configured SPF, DKIM and DMARC (My IP server is in my SPF txt).
I made a light script PHP to see if the was a problem with my DNS configuration or my server (Infomaniak) but it works well with Leantime librairies.
Here is the script I use in "public" : 
```
<?php

//Import PHPMailer classes into the global namespace
//These must be at the top of your script, not inside a function
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

//Load Composer's autoloader
require '../vendor/autoload.php';

//Create an instance; passing `true` enables exceptions
$mail = new PHPMailer(true);

try {
    //Server settings
    $mail->SMTPDebug = SMTP::DEBUG_SERVER;                      //Enable verbose debug output
    $mail->isSMTP();                                                            //Send using SMTP
    $mail->Host       = 'mail.infomaniak.com';                      //Set the SMTP server to send through
    $mail->SMTPAuth   = true;                                           //Enable SMTP authentication
    $mail->Username   = 'mailer@domain.com';                 //SMTP username
    $mail->Password   = 'xxxxxxxxxxxx';                               //SMTP password
    $mail->SMTPSecure = PHPMailer::ENCRYPTION_SMTPS;            //Enable implicit TLS encryption
    $mail->Port       = 465;                                    //TCP port to connect to; use 587 if you have set `SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS`

    //Recipients
    $mail->setFrom('mailer@domain.com', 'Mailer');
    $mail->addAddress('surname@domain2.com', 'Surname');     //Add a recipient
    $mail->addReplyTo('mailer@domain.com', 'Information');

    //Content
    $mail->isHTML(true);                                  //Set email format to HTML
    $mail->Subject = 'Here is the subject';
    $mail->Body    = 'This is the HTML message body <b>in bold!</b>';
    $mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

    $mail->send();
    echo 'Message has been sent';
} catch (Exception $e) {
    echo "Message could not be sent. Mailer Error: {$mail->ErrorInfo}";
}
?>

```
If you have any other way to find how to solve my problem, it would be great ;o)

**Comment by monkeyotg** [comment]
Following is my experience, hope it would be helpful for someone:

I am using Docker Desktop on Windows Server 2022.
I have to add DNS Configuration in the settings, otherwise Leantime Container can not resolve the domain in it's Container.
After I made this change, SMTP works OK!

![Snipaste_2024-03-13_21-28-24](https://github.com/Leantime/leantime/assets/8792193/cb2e6232-fef5-498f-ad3c-f8bcaa234845)

**Comment by EncodedBird** [comment]
It seems related to  https://github.com/PHPMailer/PHPMailer although I can't tell if its an upstream bug or not. It waits for a response from the server like it was doing starttls even though implicit TLS is being used. This is why switching to 587 works.

**Reply by refadon** [reply]
I can confirm that sending the message requires **STARTTLS** mode. Importantly, the logical parameters (**_true_** / **_false_**) in the .env file must be specified in single quotes.

## Email
LEAN_EMAIL_RETURN='leantime@domain.tld'         # Valid return email address
LEAN_EMAIL_USE_SMTP='true'                      # true -> SMTP, false -> php mail()
LEAN_EMAIL_SMTP_HOSTS='mail.domain.tld'         # SMTP host
LEAN_EMAIL_SMTP_AUTH='true'                     # true -> SMTP authentication required; email and password needed. false -> authentication not requ
LEAN_EMAIL_SMTP_USERNAME='leantime@domain.tld'  # SMTP username
LEAN_EMAIL_SMTP_PASSWORD='..........'           # SMTP password
LEAN_EMAIL_SMTP_AUTO_TLS='false'                # SMTP Enable TLS encryption automatically if a server supports it
LEAN_EMAIL_SMTP_SECURE='STARTTLS'               # SMTP Security protocol (usually one of: TLS, SSL, STARTTLS)
LEAN_EMAIL_SMTP_SSLNOVERIFY='false'             # SMTP Allow insecure SSL: Don't verify certificate, accept self-signed, etc.
LEAN_EMAIL_SMTP_PORT='587'                      # Port (usually one of 25, 465, 587, 2526)

**Comment by romifirdauslazuardi** [comment]
Anyone use Nginx Proxy Manager + Mailu?

how to setup Email for LeanTime?

log mailu smtp: 

12 19:27:50 server postfix/anvil[1917]: statistics: max connection rate 1/60s for (smtp:192.168.203.6) at Aug 12 19:24:25
Aug 12 19:27:50 server postfix/anvil[1917]: statistics: max connection count 1 for (smtp:192.168.203.6) at Aug 12 19:24:25
Aug 12 19:27:50 server postfix/anvil[1917]: statistics: max cache size 1 at Aug 12 19:24:25
Aug 12 20:34:50 server postfix/smtpd[2889]: connect from mailu-front-1.mailu_default[192.168.203.6]
INFO:root:Connect
INFO:root:Connect
Aug 12 20:34:50 server postfix/smtpd[2889]: DDF79360134: client=unknown[50.31.205.23]
INFO:root:Connect
Aug 12 20:34:51 server postfix/cleanup[2894]: DDF79360134: message-id=<851bfadc-370b-43a2-94f0-04b2c121411e@mtasv.net>
Aug 12 20:34:53 server postfix/qmgr[353]: DDF79360134: from=<SRS1=eb0l=mail.lazuardigroup.com==6HG2=2Y=pm-bounces.mail.cursor.com=pm_bounces@mail.lazuardigroup.com>, size=4841, nrcpt=1 (queue active)
Aug 12 20:34:54 server postfix/smtpd[2889]: disconnect from unknown[50.31.205.23] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=1 quit=1 commands=6/7
Aug 12 20:34:54 server postfix/lmtp[2895]: DDF79360134: to=<developer@lazuardigroup.com>, relay=front[192.168.203.6]:2525, delay=3.8, delays=3.4/0.05/0.05/0.33, dsn=2.0.0, status=sent (250 2.0.0 <developer@lazuardigroup.com> 6P9ONG2lm2iMCwAAp4iz9g Saved)
Aug 12 20:34:54 server postfix/qmgr[353]: DDF79360134: removed
Aug 12 20:38:14 server postfix/anvil[2892]: statistics: max connection rate 1/60s for (smtp:192.168.203.6) at Aug 12 20:34:50
Aug 12 20:38:14 server postfix/anvil[2892]: statistics: max connection count 1 for (smtp:192.168.203.6) at Aug 12 20:34:50
Aug 12 20:38:14 server postfix/anvil[289

log smtp front 1 :

S handshaking: read(size=596) failed: Connection reset by peer
Aug 13 05:40:36 imap-login: Info: Disconnected: Connection closed: SSL_accept() failed: error:0A0000C1:SSL routines::no shared cipher (no auth attempts in 0 secs): user=<>, rip=212.102.40.218, rport=35900, lip=192.168.203.6, lport=993, TLS handshaking: SSL_accept() failed: error:0A0000C1:SSL routines::no shared cipher
Aug 13 05:40:44 imap-login: Info: Disconnected: Connection closed: SSL_accept() failed: error:0A00006C:SSL routines::bad key share (no auth attempts in 5 secs): user=<>, rip=212.102.40.218, rport=40476, lip=192.168.203.6, lport=993, TLS handshaking: SSL_accept() failed: error:0A00006C:SSL routines::bad key share
2025/08/13 05:49:11 [info] 52#52: *1888 client 77.90.185.54:56355 connected to 0.0.0.0:25
2025/08/13 05:49:13 [info] 52#52: *1888 client login failed: "AUTH not supported" while in http auth state, client: 77.90.185.54, server: 0.0.0.0:25, login: "Administrator@lazuardigroup.com"
2025/08/13 05:51:43 [info] 52#52: *1897 client 185.196.10.175:54398 connected to 0.0.0.0:25
2025/08/13 05:51:46 [info] 52#52: *1897 client login failed: "AUTH not supported" while in http auth state, client: 185.196.10.175, server: 0.0.0.0:25, login: "luck"
2025/08/13 05:59:28 [info] 52#52: *1916 client 77.90.185.54:51813 connected to 0.0.0.0:25
2025/08/13 05:59:29 [info] 52#52: *1916 client login failed: "AUTH not supported" while in http auth state, client: 77.90.185.54, server: 0.0.0.0:25, login: "AdvWebadmin@lazuardigroup.com"
2025/08/13 06:09:39 [info] 52#52: *1941 client 77.90.185.54:64213 connected to 0.0.0.0:25
2025/08/13 06:09:40 [info] 52#52: *1941 client login failed: "AUTH not supported" while in http auth state, client: 77.90.185

(some hackers trying to use port 25)

FYI: when i try connection mailu to LeanTime its only open port 25 & 465. 587 no output.

**Reply by Yaruss** [reply]
If you need to send emails through Mailu, you should specify LEAN_EMAIL_SMTP_SECURE='ssl' in lowercase and port 465 in the configuration.

**Comment by developersteve** [comment]
Not sure if this is documented anywhere, and there is no UI error surfaced to display this, but theres a invite user timeout of 4 minutes between invite sends. So be wary if you use invite user to do some SMTP testing.

**Comment by verybigelephants** [comment]
for anyone coming here from google - it is "resend invite" button that did nothing for me, there is some bug. after i created new user and sent an email then it worked normally
